website/content/api-docs/secret/pki.mdx

```diff
@@ -1345,6 +1345,203 @@ $ curl \
 }
 ```
 
+### List CEL roles
+
+This endpoint returns a list of available CEL roles. Only the role names are
+returned, not any values. It is useful to both operators and users.
+
+| Method | Path         |
+| :----- | :----------- |
+| `LIST` | `/pki/cel/roles` |
+
+#### Parameters
+
+ - `after` `(string: "")` - Optional entry to begin listing after for
+   pagination; not required to exist.
+
+ - `limit` `(int: 0)` - Optional number of entries to return; defaults
+   to all entries.
+
+#### Sample request
+
+```shell-session
+$ curl \
+    --header "X-Vault-Token: ..." \
+    --request LIST \
+    http://127.0.0.1:8200/v1/pki/cel/roles
+```
+
+#### Sample response
+
+```json
+{
+  "request_id":"8aa74f79-6e7f-9157-38fd-2a66b1be9071",
+  "lease_id":"",
+  "renewable":false,
+  "lease_duration":0,
+  "data":{"keys":["testrole"]},
+  "wrap_info":null,
+  "warnings":null,
+  "auth":null
+}
+```
+
+### Read CEL role
+
+This endpoint queries the CEL role definition. It is useful to both operators and
```

Comment:
I know this is boilerplate from the regular role read endpoints, but I don't know how true this is. It is hard to imagine most callers finding value in these, and it might arguably be a security risk to disclose validation logic.

@fatima2003 @suprjinx Not for now, but I wonder if it is worthwhile to add a `description` to CEL roles, to allow people space to add a human-readable description of the relevant statement? Probably for both programs and variables fwiw. In the future, we could perhaps add a `cel/role/<role>/description` endpoint, which just returns the populated description fields, perhaps?